requirements/fragment.py

```diff
@@ -27,3 +27,10 @@ def get_hash_info(d):
             return key, d[key]
 
     return None, None
+
+
+def parse_extras_require(egg):
+    if egg is not None and ('[' and ']') in egg:
```

Comment:
This line is not quite right:

    ('[' and ']') in 'eggblah]'  # True

Possibly just use a regex?